### ISSUE TITLE
Error handling best practices

### ISSUE THREAD

**Issue by NicholasGWK** [kafkajs/confluent-schema-registry#81]
Hello! This is a combination of a request for some guidance of how the lib works/ additional documentation. Happy to help with the latter once I understand things a bit better.

Essentially, I'm using this lib in my kafka consumer to decode messages, which works great. But now that I'm running through the failure cases, I don't see anything about the types of errors that can be thrown and how to handle them. 

Perusing the code it seems like there are two classes of errors that can possibly happen:

- API calls to confluent fail, possibly due to bad creds or transient errors.
- The underlying avsc library fails to parse a message and throws an error once the schema is fetch/grabbed from the cache etc.

So in my implementation I'm:

- Consuming a message from the topic with kafkaJS
- Fetching the schema
- Decoding the message in the consumer

My questions are:

- Is this the correct way to use the lib?
- Is it possible for `registry.decode(message)` to fail provided the schema is fetched correctly?
- Did I read the code right when I say I don't think any of the internal error types are exposed and that any fetching/decoding error will just end up bubbling up as a regular error?

Basically I'm looking to solve the issue of "should I retry fetching the schema/decoding this message, or is it a lost cause if it's the decoding that failed or my creds are bad", but based on my second question this might be unnecessary.

Thanks a lot for the hard work on the lib, and thanks in advance for the help!

Cheers,

Nick
